google/resource_compute_instance_test.go

```diff
@@ -1948,6 +1984,32 @@ func testAccCheckComputeInstanceHasStatusTerminated(instance *compute.Instance)
 	}
 }
 
+func testAccCheckComputeInstanceStop(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*Config)
+
+		op, err := config.clientCompute.Instances.Stop(config.Project, rs.Primary.Attributes["zone"], rs.Primary.ID).Do()
+		if err != nil {
+			return fmt.Errorf("Could not stop instance: %s", err)
+		}
+		err = computeOperationWait(config.clientCompute, op, config.Project, "Waiting on stop")
+		if err != nil {
+			return fmt.Errorf("Could not stop instance: %s", err)
+		}
+
```

Comment:
I used `testAccCheckComputeInstanceExistsInProject` as an example. I don't know if I should pass also here a `instance *compute.Instance` parameter and change its status (`instance.Status = "TERMINATED"`) before returning. It will simplify the acceptance test `TestAccComputeInstance_statusTerminatedManuallyAndRestart` (see my comment there).